ENH: Replaced --make-shell-scripts by '--make-scripts shell' or 'batch'

N-Dekker opened the pull request:
make_registration_scripts now supports '--make-scripts shell' for Unix .sh files, and '--make-scripts batch' for Windows .bat files.

Implementation includes suggestion by Jirka Borovec (@Borda)

N-Dekker committed:
ENH: Replaced --make-shell-scripts by '--make-scripts shell' or 'batch'

make_registration_scripts now supports '--make-scripts shell' for Unix .sh files, and '--make-scripts batch' for Windows .bat files.

Implementation includes suggestion by Jirka Borovec (@Borda)

N-Dekker commented:
Please have a look! I would like to merge this pull request tomorrow (Thursday).

kaspermarstal reviewed (approved):
Looking good. Might be a stupid question but just to be sure: Have you tested that the scripts are executable on both Windows and Linux?

N-Dekker commented:
> Have you tested that the scripts are executable on both Windows and Linux?

Something seems wrong in my local Windows test, still need to figure out. Regarding Linux, is that tested automatically, when the commit is merged?

kaspermarstal commented:
We don't run the Python scripts as part of the CI but it is tested via the challenge.

N-Dekker commented:
> We don't run the Python scripts as part of the CI but it is tested via the challenge.

Then I guess this pull request would break the challenge, because it would no longer support the old `--make-shell-scripts` syntax. Is that OK?

N-Dekker commented:
I hesitate now if we shouldn't preserve backward compatibility. Couldn't we just keep supporting that one option `--make-shell-scripts`, and then `if platform.system() == "Windows": script_file_extension = ".bat"`?

kaspermarstal commented:
That is a good idea but it should still be configurable. For example, these days there are many ways to run shell scripts on Windows, so perhaps users would like to use shell scripts even though they are on Windows. Thoughts?

N-Dekker commented:
> That is a good idea but it should still be configurable. For example, these days there are many ways to run shell scripts on Windows, so perhaps users would like to use shell scripts even though they are on Windows. Thoughts?

We can always make it configurable afterwards. For now, I would only want to implement this option (".sh" for Windows) _if_ there is a user who actually needs that feature. OK?

kaspermarstal commented:
OK!

N-Dekker committed:
ENH: Restored --make-shell-scripts support

Now using ".bat" file extension if platform is 'Windows'. Discussed with @kaspermarstal
